autogpt_platform/backend/backend/blocks/http.py

```diff
@@ -169,3 +214,61 @@ async def run(
             yield "client_error", result
         else:
             yield "server_error", result
+
+
+class SendAuthenticatedWebRequestBlock(SendWebRequestBlock):
+    class Input(SendWebRequestBlock.Input):
+        credentials: HttpCredentials = SchemaField(
+            description="HTTP host-scoped credentials for automatic header injection"
+        )
+
+    def __init__(self):
+        Block.__init__(
+            self,
+            id="7f4f8b2a-c3d1-4e5f-8a9b-2c3d4e5f6a7b",
+            description="Make an authenticated HTTP request with host-scoped credentials (JSON / form / multipart).",
+            categories={BlockCategory.OUTPUT},
+            input_schema=SendAuthenticatedWebRequestBlock.Input,
+            output_schema=SendWebRequestBlock.Output,
+            test_credentials=TEST_CREDENTIALS,
+        )
+
+    async def run(  # type: ignore[override]
```

Comment:
??

Reply:
The parent doesn't have credentials as part of the param

Reply:
Also the input type is different
```
Method "run" overrides class "SendWebRequestBlock" in an incompatible manner
  Parameter 2 type mismatch: base parameter is type "Input", override parameter is type "Input"
    "backend.blocks.http.SendWebRequestBlock.Input" is incompatible with "backend.blocks.http.SendAuthenticatedWebRequestBlock.Input"
```
idk why it's incompatible even though the later one extending the former